.github/workflows/retest.yaml

```diff
@@ -0,0 +1,24 @@
+name: Detect and Trigger Retest
+on:
+  issue_comment:
+    types: [created, edited]
+
+permissions:
+  contents: read
+
+jobs:
+  retest:
+    runs-on: ubuntu-latest
```

Comment:
conventionally we put this farther down just before `steps`, as it's less important than `name`, `if`, `permissions`. see also #11752